From a91b0ad8df732cf7976e8a665de11e3f0f50a9da Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <julius.kuenzel@kde.org>
Date: Sun, 17 Nov 2024 09:28:46 +0100
Subject: [PATCH 1/2] Make qrc generation fail if no *.theme file was found

Without any theme file the icon theme will not work at runtime
---
 src/tools/qrcAlias.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/tools/qrcAlias.cpp b/src/tools/qrcAlias.cpp
index 0ddfe1171..f4d1de38f 100644
--- a/src/tools/qrcAlias.cpp
+++ b/src/tools/qrcAlias.cpp
@@ -117,6 +117,7 @@ static void generateQRCAndCheckInputs(const QStringList &indirs, const QString &
     // loop over the inputs, remember if we do look at generated stuff for checks
     bool generatedIcons = false;
     QSet<QString> checkedFiles;
+    bool themeFileFound = false;
     for (const auto &indir : indirs) {
         // go to input dir to have proper relative paths
         if (!QDir::setCurrent(indir)) {
@@ -180,6 +181,8 @@ static void generateQRCAndCheckInputs(const QStringList &indirs, const QString &
                 if (!generatedIcons) {
                     checkForDuplicates(fullPath);
                 }
+            } else if (fullPath.endsWith(QLatin1String(".theme"))) {
+                themeFileFound = true;
             }
 
             // write the one alias to file entry
@@ -190,6 +193,11 @@ static void generateQRCAndCheckInputs(const QStringList &indirs, const QString &
         generatedIcons = true;
     }
 
+    if (!themeFileFound) {
+        // without any theme file the icon theme will not work at runtime
+        qFatal() << "No theme file found!";
+    }
+
     out.write("</qresource>\n");
     out.write("</RCC>\n");
 }
-- 
GitLab


From 4e8d083af4df45c683f04498f8750d593a63e763 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <julius.kuenzel@kde.org>
Date: Sun, 17 Nov 2024 09:54:26 +0100
Subject: [PATCH 2/2] Generate index.theme unconditionally to fix qrc/rcc

Related to c39caf44b2e20a76b7d6f70735b203b17d1b85a0 which broke the icons eg. on Windows
---
 icons-dark/CMakeLists.txt | 9 ++++++---
 icons/CMakeLists.txt      | 9 ++++++---
 2 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/icons-dark/CMakeLists.txt b/icons-dark/CMakeLists.txt
index 0a065c803..567a7b1c2 100644
--- a/icons-dark/CMakeLists.txt
+++ b/icons-dark/CMakeLists.txt
@@ -33,6 +33,12 @@ else()
     )
 endif()
 
+# index.theme is used by the lib (via qrc/rcc)
+# and gets installed in the condition below
+file(COPY_FILE index.theme.in index.theme)
+file(READ ../commonthemeinfo.theme.in COMMON_THEME_INFO)
+file(APPEND index.theme ${COMMON_THEME_INFO})
+
 if(NOT SKIP_INSTALL_ICONS)
     install(DIRECTORY ${breeze_icon_dark_dirs} DESTINATION ${BREEZE_INSTALL_DIR})
     file(GLOB_RECURSE icon_files RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "${CMAKE_CURRENT_SOURCE_DIR}/*.svg")
@@ -43,9 +49,6 @@ if(NOT SKIP_INSTALL_ICONS)
         endif()
     endforeach()
 
-    file(COPY_FILE index.theme.in index.theme)
-    file(READ ../commonthemeinfo.theme.in COMMON_THEME_INFO)
-    file(APPEND index.theme ${COMMON_THEME_INFO})
     install(FILES index.theme DESTINATION ${BREEZE_INSTALL_DIR})
 
     if(NOT WIN32)
diff --git a/icons/CMakeLists.txt b/icons/CMakeLists.txt
index 0049d7169..787d780ae 100644
--- a/icons/CMakeLists.txt
+++ b/icons/CMakeLists.txt
@@ -34,15 +34,18 @@ else()
     )
 endif()
 
+# index.theme is used by the lib (via qrc/rcc)
+# and gets installed in the condition below
+file(COPY_FILE index.theme.in index.theme)
+file(READ ../commonthemeinfo.theme.in COMMON_THEME_INFO)
+file(APPEND index.theme ${COMMON_THEME_INFO})
+
 if(NOT SKIP_INSTALL_ICONS)
     install(DIRECTORY ${breeze_icon_dirs} DESTINATION ${BREEZE_INSTALL_DIR})
     # collect all our icons we need to install, will be used by dark icon set as fallback icons if not there, too
     file(GLOB_RECURSE icon_files RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "${CMAKE_CURRENT_SOURCE_DIR}/*.svg")
     set(BREEZE_LIGHT_ICONS ${icon_files} PARENT_SCOPE)
 
-    file(COPY_FILE index.theme.in index.theme)
-    file(READ ../commonthemeinfo.theme.in COMMON_THEME_INFO)
-    file(APPEND index.theme ${COMMON_THEME_INFO})
     install(FILES index.theme DESTINATION ${BREEZE_INSTALL_DIR})
 
     if(NOT WIN32)
-- 
GitLab

